src/Task4/Main.java

```diff
@@ -0,0 +1,33 @@
+package Task4;
+
+public class Main {
+    public static void main(String[] args) {
+
+        ListOfNumbers listObject1 = new ListOfNumbers();
+        listObject1.addNumber(1.5d);
+        listObject1.addNumber(10d);
+        listObject1.addNumber(2.5d);
+        ListOfNumbers listObject2 = new ListOfNumbers();
+        listObject2.addNumber(0d);
+        listObject2.addNumber(1d);
+        ListOfNumbers listObject3 = new ListOfNumbers();
+        listObject3.addNumber(100d);
+
+        ListOfNumbers result1 = findWithMaxAverage(listObject1,listObject2,listObject3);
+        ListOfNumbers result2 = findWithMaxAverage(listObject1,listObject2);
+
+        System.out.println(result1);
+        System.out.println(result2);
+
+    }
+
+    private static ListOfNumbers findWithMaxAverage(ListOfNumbers...listObject) {
+        ListOfNumbers maxAverage = listObject[0];
+        for (ListOfNumbers l:listObject){
+            if (l.getAverage() > maxAverage.getAverage()){
```

Comment:
`maxAverage` saugočiau kaip skaičių, o ne kaip ListOfNumbers objektą. 

Įsivaizduok, kad vidurkio skaičiavimas užtrunka kokių 10s. Kiekvienos ciklo iteracijos metu, patikrinimas užtrunka kaip minimum 20s, nes apskaičiuoji vienos ir kitos pusės vidurkį. Jeigu maxAverage objektas nepasikeis - 100 elementų atveju - 100 kartų skaičiuotum to pačio maxAverage objekto vidurkį - veltui pralaukta 1000s - 16min